recipes/fsx_mount.rb

```diff
@@ -0,0 +1,81 @@
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: fsx_mount
+#
+# Copyright 2013-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+fsx_shared_dir = node['cfncluster']['cfn_fsx_shared_dir'].split(',')[0]
+fsx_fs_id = node['cfncluster']['cfn_fsx']
+
+lustre_kmod_rpm = "#{node['cfncluster']['sources_dir']}/kmod-lustre-client-#{node['cfncluster']['lustre']['version']}.x86_64.rpm"
+lustre_client_rpm = "#{node['cfncluster']['sources_dir']}/lustre-client-#{node['cfncluster']['lustre']['version']}.x86_64.rpm"
+
+# Check to see if FSx is created
+if fsx_shared_dir != "NONE"
+
+  # Path needs to be fully qualified, for example "shared/temp" becomes "/shared/temp"
+  if !fsx_shared_dir.start_with?("/")
+    fsx_shared_dir = "/" + fsx_shared_dir
+  end
+
+  # Create the shared directories
+  directory fsx_shared_dir do
+    owner 'root'
+    group 'root'
+    mode '1777'
+    action :create
+  end
+
+  # Get Lustre Kernel Module RPM
+  remote_file lustre_kmod_rpm do
+    source node['cfncluster']['lustre']['kmod_url']
+    mode '0644'
+    retries 3
+    retry_delay 5
+    not_if { ::File.exist?(lustre_kmod_rpm) }
+    action :create_if_missing
+  end
+
+  # Get Lustre Client RPM
+  remote_file lustre_client_rpm do
+    source node['cfncluster']['lustre']['client_url']
+    mode '0644'
+    retries 3
+    retry_delay 5
+    not_if { ::File.exist?(lustre_client_rpm) }
+    action :create_if_missing
+  end
+
+  # Install lustre mount drivers
+  yum_package 'lustre_kmod' do
+    source lustre_kmod_rpm
+  end
+
+  # Install lustre mount drivers
+  yum_package 'lustre_client' do
+    source lustre_client_rpm
+  end
+
+  # Mount FSx over NFS
+  mount fsx_shared_dir do
+    device "#{fsx_fs_id}.fsx.#{node['cfncluster']['cfn_region']}.amazonaws.com:/fsx"
```

Comment:
since fsx_fs_id is used only here, you can use node['cfncluster']['cfn_fsx'] directly

Reply:
changed